lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -88,6 +89,10 @@ def configure(conf)
       @remove_keys_on_update = @remove_keys_on_update.split ','
     end
 
+    if @logstash_prefix_key && @logstash_prefix_key.is_a?(String)
+      @logstash_prefix_key = @logstash_prefix_key.split '.'
```

Comment:
depending upon which element you need of the array you could also use 'first', 'last'